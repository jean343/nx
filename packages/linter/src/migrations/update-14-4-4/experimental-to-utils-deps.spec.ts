import { updateJson } from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import { WORKSPACE_PLUGIN_DIR } from '../../generators/workspace-rules-project/workspace-rules-project';
import addTypescriptEslintUtilsIfNeeded from './experimental-to-utils-deps';

describe('addTypescriptEslintUtilsIfNeeded()', () => {
  it('should remove @typescript-eslint/experimental-utils from package.json devDependencies and add @typescript-eslint/utils', async () => {
    const tree = createTreeWithEmptyWorkspace();
    updateJson(tree, 'package.json', (json) => {
      json.devDependencies['@typescript-eslint/experimental-utils'] = '12.3.4';
      return json;
    });

    await addTypescriptEslintUtilsIfNeeded(tree);

    expect(tree.read(`package.json`).toString('utf-8')).toMatchSnapshot();
  });

  it('should remove @typescript-eslint/experimental-utils from package.json devDependencies and add @typescript-eslint/utils', async () => {
    const tree = createTreeWithEmptyWorkspace();
    updateJson(tree, 'package.json', (json) => {
      json.dependencies['@typescript-eslint/experimental-utils'] = '12.3.4';
      return json;
    });

    await addTypescriptEslintUtilsIfNeeded(tree);

    expect(tree.read(`package.json`).toString('utf-8')).toMatchSnapshot();
  });

  it('should add @typescript-eslint/utils if plugins folder exists', async () => {
    const tree = createTreeWithEmptyWorkspace();

    tree.write(
      `${WORKSPACE_PLUGIN_DIR}/rules/my-rule.ts`,
      'console.log("not important");'
    );

    await addTypescriptEslintUtilsIfNeeded(tree);

    expect(tree.read(`package.json`).toString('utf-8')).toMatchSnapshot();
  });

  it('should not add @typescript-eslint/utils if plugins folder or experimental-utils dont exist', async () => {
    const tree = createTreeWithEmptyWorkspace();

    await addTypescriptEslintUtilsIfNeeded(tree);

    expect(tree.read(`package.json`).toString('utf-8')).toMatchSnapshot();
  });
});
