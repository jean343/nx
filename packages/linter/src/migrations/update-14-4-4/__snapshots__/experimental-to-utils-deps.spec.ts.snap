// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addTypescriptEslintUtilsIfNeeded() should add @typescript-eslint/utils if plugins folder exists 1`] = `
"{
  \\"name\\": \\"test-name\\",
  \\"dependencies\\": {},
  \\"devDependencies\\": {
    \\"@typescript-eslint/utils\\": \\"^5.29.0\\"
  }
}
"
`;

exports[`addTypescriptEslintUtilsIfNeeded() should not add @typescript-eslint/utils if plugins folder or experimental-utils dont exist 1`] = `"{\\"name\\":\\"test-name\\",\\"dependencies\\":{},\\"devDependencies\\":{}}"`;

exports[`addTypescriptEslintUtilsIfNeeded() should remove @typescript-eslint/experimental-utils from package.json devDependencies and add @typescript-eslint/utils 1`] = `
"{
  \\"name\\": \\"test-name\\",
  \\"dependencies\\": {},
  \\"devDependencies\\": {
    \\"@typescript-eslint/utils\\": \\"^5.29.0\\"
  }
}
"
`;

exports[`addTypescriptEslintUtilsIfNeeded() should remove @typescript-eslint/experimental-utils from package.json devDependencies and add @typescript-eslint/utils 2`] = `
"{
  \\"name\\": \\"test-name\\",
  \\"dependencies\\": {},
  \\"devDependencies\\": {
    \\"@typescript-eslint/utils\\": \\"^5.29.0\\"
  }
}
"
`;
